rasa/core/events/__init__.py

```diff
@@ -98,38 +110,42 @@ def from_story_string(
         parameters: Dict[Text, Any],
         default: Optional[Type["Event"]] = None,
     ) -> Optional[List["Event"]]:
-        event = Event.resolve_by_type(event_name, default)
+        event_class = Event.resolve_by_type(event_name, default)
 
-        if event:
-            return event._from_story_string(parameters)
-        else:
+        if not event_class:
             return None
 
+        return event_class._from_story_string(parameters)
+
     @staticmethod
     def from_parameters(
         parameters: Dict[Text, Any], default: Optional[Type["Event"]] = None
     ) -> Optional["Event"]:
 
         event_name = parameters.get("event")
-        if event_name is not None:
-            copied = parameters.copy()
-            del copied["event"]
+        if event_name is None:
+            return None
 
-            event = Event.resolve_by_type(event_name, default)
-            if event:
-                return event._from_parameters(parameters)
-            else:
-                return None
-        else:
+        copied = parameters.copy()
+        del copied["event"]
+
+        event_class = Event.resolve_by_type(event_name, default)
+        if not event_class:
             return None
 
+        return event_class._from_parameters(parameters)
```

Comment:
Why are we copying and deleting from `parameters` and then returning the original?

Reply:
Excellent catch!! Looks like that's a leftover from 48fb141de93397be44c4de5dcd52f15a00b05f36. I'll remove it.